networkit/cpp/graph/Graph.cpp

```diff
@@ -1065,4 +1068,52 @@ Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes) const {
 	return S;
 }
 
+void Graph::toLargestConnectedComponent() {
+	if (this->isDirected())
+		throw std::runtime_error(
+			"Method available for undirected graphs only.");
+
+	ConnectedComponents cc(*this);
+	cc.run();
+	auto comps = cc.getComponents();
+	std::sort(comps.begin(), comps.end(),
+		[&](const std::vector<node> &x, const std::vector<node> &y) {
+			return x.size() > y.size();
+	});
+	std::unordered_map<node, node> map;
+	std::unordered_set<node> set(comps[0].begin(), comps[0].end());
```

Comment:
Why are you storing the nodes in an unordered set? You can already:
* Iterate over the nodes in the component in `comps[0]`
* Get the component id of any of these nodes and compare it to the connected component of any node to check if that other node is in the largest connected component.

The first is in linear time, the second in constant time, there is no need to use a potentially large and inefficient `unordered_set`.